### ISSUE TITLE
Add Snakemake workflow for training experiments

### ISSUE THREAD

**Issue by ntatonetti** [tatonetti-lab/onsides#81]
The old experiment orchestration system (`experiment_tracker.py` + `experiments.json`) was removed during the v2→v3 refactoring. Training is now handled by `onsides-train` (added in the #30 work), which runs a single training job per invocation.

To support hyperparameter sweeps (e.g. #25, learning rate tuning for Boxed Warnings), we need an experiment orchestration layer. Since the rest of the pipeline uses Snakemake, the natural approach is a new Snakefile that parameterizes training runs.

### Proposed approach

Add `snakemake/onsides/train/Snakefile` that:
- Defines experiment configurations as Snakemake `expand()` over hyperparameter grids (learning rate, section, nwords, etc.)
- Calls `onsides-train` for each configuration
- Collects epoch results CSVs into a summary for comparison
- Supports `--config` overrides so users can run subsets of the grid

### Example usage
```bash
# Run all experiments
snakemake -s snakemake/onsides/train/Snakefile

# Run only BW learning rate sweep
snakemake -s snakemake/onsides/train/Snakefile --config sections=["BW"] learning_rates=["1e-6","2.5e-5","1e-5"]
```

### Context
- `onsides-train` CLI: `src/onsides/train.py` (supports `--resume` for interrupted runs)
- Old experiment system: `v2` branch (`src/experiment_tracker.py`, `experiments.json`)
- Related: #25 (tune LR for BW), #31 (new reference set evaluation)
